packages/rpc/src/common.ts

```diff
@@ -50,27 +51,62 @@ type AdditionalOrder = {
   };
 };
 
+type CcmParams = {
+  message: HexString;
+  gas_budget: HexString;
+};
+
+type FillOrKillParams = {
+  refund_address: string;
+  retry_duration: number;
+  min_price: `0x${string}`;
+};
+
+type DcaParams = {
+  number_of_chunks: number;
+  chunk_interval: number;
+};
+
 export type RpcRequest = WithHash<{
   broker_requestSwapDepositAddress: [
     sourceAsset: UncheckedAssetAndChain,
     destinationAsset: UncheckedAssetAndChain,
     destinationAddress: string,
     brokerCommission: number,
-    ccmMetadata?: Nullish<{
-      message: HexString;
-      gas_budget: HexString;
-    }>,
+    ccmMetadata?: Nullish<CcmParams>,
+    boostFee?: Nullish<number>,
+    affiliateFees?: Nullish<{ account: string; bps: number }[]>,
+    fillOrKillParams?: Nullish<FillOrKillParams>,
+    dcaParams?: Nullish<DcaParams>,
+  ];
+  broker_request_swap_parameter_encoding: [
+    sourceAsset: UncheckedAssetAndChain,
+    destinationAsset: UncheckedAssetAndChain,
+    destinationAddress: string,
+    brokerCommission: number,
+    extraParameters:
+      | {
+          chain: 'Bitcoin';
+          min_output_amount: `0x${string}`;
+          retry_duration: number;
```

Comment:
not fokParams 😕  ?

Reply:
is that because of size constraints? or why is it encoded differently for btc?

Reply:
yes i think for btc its important to keep the size low